views/users.py

```diff
@@ -85,8 +85,14 @@ def user_password_forgot():
 def user_password_reset(code):
     reset_form = PasswordResetForm()
     if reset_form.validate_on_submit():
-        # TODO
-        pass
+        token = PasswordResetToken.query.filter_by(token=reset_form.code.data).first()
+        user = User.query.filter(func.lower(User.email) == func.lower(token.email)).first()
```

Comment:
token.email.lower() instead of func.lower(token.email); token.email is a constant string and not a dynamic sql object